corehq/apps/enterprise/tests/test_apis.py

```diff
@@ -0,0 +1,111 @@
+        form2 = self._create_form(mobile_user_2, form_id='2345', received_on=datetime(year=2004, month=10, day=12))
+        form_adapter.bulk_index([form1, form2], refresh=True)
+
+        request = self._create_api_request(enterprise_admin, 'test-domain-1', api_key)
+
+        resource = FormSubmissionResource()
+        response = resource.dispatch_list(request)
+        result = json.loads(response.content)
+        resulting_forms = result['value']
+        self.assertEqual([form['form_id'] for form in resulting_forms], ['1234', '2345'])
+
+    def _create_form(self, user, form_id=None, received_on=None):
+        form_data = {
+            '#type': 'fake-type',
+            '@name': 'TestForm',
+            'meta': {
+                'userID': user._id,
+                'username': user.name,
+                'instanceID': form_id,
+            },
+        }
+        return create_form_for_test(
+            user.domain,
+            user_id=user._id,
+            form_data=form_data,
+            form_id=form_id,
+            received_on=received_on,
+        )
+
+    def _create_enterprise_account_covering_domains(self, domains):
+        billing_account = generator.billing_account('test-admin@dimagi.com', 'test-admin@dimagi.com')
+        billing_account.enterprise_admin_emails = ['test-admin@dimagi.com']
+        billing_account.save()
+
+        for domain in domains:
+            domain_obj = Domain(name=domain, is_active=True)
+            domain_obj.save()
+            self.addCleanup(domain_obj.delete)
+
+            generator.generate_domain_subscription(
+                billing_account, domain_obj, datetime.now(timezone.utc), None, is_active=True
+            )
+
+        return billing_account
+
+    def _create_enterprise_admin(self, email, domain):
+        user = WebUser.create(
+            domain, email, 'test123', None, None, email)
+        user.is_superuser = True
+        user.save()
+
+        self.addCleanup(user.delete, None, deleted_by=None)
+
+        role = Role.objects.create(slug="test_role")
+        UserRole.objects.create(user=user.get_django_user(), role=role)
+        accounting_admin_role = Role.objects.get_or_create(
+            name="Accounting Admin",
+            slug=privileges.ACCOUNTING_ADMIN,
+        )[0]
+        Grant.objects.create(from_role=role, to_role=accounting_admin_role)
+
+        return user
+
+    def _create_mobile_user(self, username, domain):
+        user = CommCareUser.create(domain, username, 'test123', None, None)
+        self.addCleanup(user.delete, None, deleted_by=None)
+        return user
+
+    def _create_api_request(self, user, domain, api_key):
+        auth_string = f'{user.username}:{api_key.key}'
+        factory = RequestFactory()
+        encoded_auth = base64.b64encode(auth_string.encode()).decode()
+        request = factory.get(
+            '/',
+            {'startdate': '2004-10-10', 'enddate': '2004-11-10'},
```

Comment:
will push to a different level

Reply:
Included as part of bff5facbf3137ff0f7d757ab19538a65d1ce7b37